### PR TITLE
codec transplantation

### PR DESCRIPTION
**Pull request by andyLiuChina**
transplant codecparsers from gstream-codecparsers, and modify some codes under decoder directory.

### CODE REVIEW

**Comment by andyLiuChina** [comment]
All your suggestion above has been implemented!

File: gst/codecparserdef.h [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+/*The maximum value which can be held in a signed char*/
+#define G_MAXINT8	((signed char)  0x7f)
+/*The maximum value which can be held in a unsigned short*/
+#define G_MAXUINT16	((unsigned short) 0xffff)
+/*The maximum value which can be held in a unsigned int*/
+#define G_MAXUINT UINT_MAX
+
+static inline uint32_t 
+g_bit_storage(uint32_t value)
```
Comment (xuguangxin) [inline gst/codecparserdef.h]: please move this to separate file too and add copyright too.

File: gst/codecparserdef.h [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+  return bits;
+}
+
+#define g_new(struct_type, n_structs)  \
+    (struct_type*)malloc(sizeof(struct_type)*(n_structs))
+
+#define g_new0(struct_type, n_structs)  \
+    ({ struct_type* tmp = (struct_type*)malloc(sizeof(struct_type)*(n_structs)); \
+    if(tmp) \
+        memset(tmp, 0, sizeof(struct_type)*(n_structs));\
```
Comment (xuguangxin) [inline gst/codecparserdef.h]: please move this to separate file too and add copyright too.

File: gst/codecparserdef.h [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        free(mem);
+}
+
+#define g_return_if_fail(expr) { \
+    if (!(expr)) \
+    { \
+        return;	\
+    }; }
+
+#define g_return_val_if_fail(expr, val)	{ \
```
Comment (xuguangxin) [inline gst/codecparserdef.h]: please move this to separate file too and add copyright too.

File: gst/codecparserdef.h [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+    if (!(expr)) \
+    { \
+        return val; \
+    }; }
+
+#define _G_BOOLEAN_EXPR(expr) \
```
Comment (xuguangxin) [inline gst/codecparserdef.h]: please move this to separate file too and add copyright too.

File: gst/codecparserdef.h [hunk 5]
```diff
@@ -0,0 +1,181 @@
+#ifndef __CODEC_PARSER_DEF__
+#define __CODEC_PARSER_DEF__
```
Comment (xuguangxin) [inline gst/codecparserdef.h]: add copyright for this file

**Comment by xuguangxin** [comment]
+1

**Comment by YuJiankang** [comment]
+1

**Pull request closed without merging**